tooltester/src/main/java/io/dockstore/tooltester/runWorkflow/WorkflowRunner.java

```diff
@@ -339,20 +394,90 @@ private Execution.ExecutionStatusEnum getExecutionStatus() {
+                            + " for this workflow, the following were discovered: " + System.lineSeparator()
+                            + ecsTasks);
+                }
+                DescribeTaskDefinitionRequest request = DescribeTaskDefinitionRequest.builder().taskDefinition(ecsTaskArn).build();
+                DescribeTaskDefinitionResponse response = ecsClient.describeTaskDefinition(request);
+                taskDefinitionFamily = response.taskDefinition().family();
+                break;
+            }
+
+            try {
+                final Long waitTime = 20L;
+                TimeUnit.SECONDS.sleep(waitTime);
+            } catch (Exception ex) {
+                exceptionMessage(ex, "time error", COMMAND_ERROR);
+            }
+        }
+    }
+
     public void uploadRunInfo() {
         List<Execution> executions = new ArrayList<>();
-        Execution execution = new Execution();
-        execution.setExecutionStatus(getExecutionStatus());
+        runMetrics.setExecutionStatus(getExecutionStatus());
 
         if (getTotalWallClockTimeInISO861Standard() != null) {
-            execution.setExecutionTime(getTotalWallClockTimeInISO861Standard());
+            runMetrics.setExecutionTime(getTotalWallClockTimeInISO861Standard());
         }
 
-        executions.add(execution);
+        executions.add(runMetrics);
 
+        addSingleMetricToQa("CpuUtilized");
+        addSingleMetricToQa("MemoryUtilized");
         extendedGa4GhApi.executionMetricsPost(executions, Partner.AGC.name(), getEntryNameForApi(), version, "generated with tooltester ('run-workflows' command)");
     }
 
+    private void addSingleMetricToQa(String metricName) {
+        CloudWatchClient cloudWatchClient = CloudWatchClient.builder().build();
+        Dimension clusterNameDimension = Dimension.builder()
+                .name("ClusterName")
+                .value(clusterName)
+                .build();
+        Dimension clusterTaskDefinitionFamily = Dimension.builder()
+                .name("TaskDefinitionFamily")
+                .value(taskDefinitionFamily)
+                .build();
+        final int period = 60;
+        GetMetricStatisticsRequest request = GetMetricStatisticsRequest.builder()
+                .namespace("ECS/ContainerInsights")
+                .metricName(metricName)
+                .dimensions(clusterNameDimension, clusterTaskDefinitionFamily)
+                .period(period)
+                .statistics(Statistic.AVERAGE)
+                .startTime(workflowStartTime.toInstant())
+                .endTime(workflowEndTime.toInstant())
+                .build();
+        GetMetricStatisticsResponse response = cloudWatchClient.getMetricStatistics(request);
+        Double sumOfDataPoints = 0D;
+        Double maxDataPoint = 0D;
+        for (Datapoint datapoint: response.datapoints()) {
+            sumOfDataPoints += datapoint.average();
```

Comment:
This not actually obtaining the average. This is because the ECS container is only collecting metrics every minute I have asked for a minute by minute metric breakdown. So we are getting the only statistic collected for each minute.

If instead I asked for the metrics only every 5 minutes, then the `datapoint.average()` would average the 5 data points collected by the ECS container in those 5 minutes.

Reply:
I added the following comment:
```
            // datapoint.average() is not actually obtaining the average. This is because the ECS container is only
            // collecting metrics every minute, and we have asked for a minute by minute metric breakdown.
            // So, we are getting the only statistic collected for each minute.
```